### PR TITLE
feat: new anonymous question creation logic

### LINKED ISSUE #143: Modify create question action.
Modify the creation of anonymous questions: 
- Select a specific person to enable the option to create anonymous question (switch)
- Redirect to the detail of the questions, once the questions has been created. 
- launch a pop-up to inform the users to save the link and that your questions is not going to appear in the home page until the person who they selected publish it. 
- set `is_public` to `false`

Add in the “Thins to keep in mind” (You can also use this in the pop-up):
 
>   - For anonymous questions, you have to select a specif person according to the department selected. 
>   - The anonymous questions will not appear in the home page until an admin publish it. 
>   - Although the question has not been published, you will have an answer.r. 
>   - You must save the link of the question.

### LINKED ISSUE #172: Fix listEmployees controller to skip employees without a fullname
During testing in https://questions-develop.wizeline.com/ the following error was found 

Image

suggesting there are employees without a full_name. A easy fix would adding to the Prisma query to skip all that do not have the required values.

### PR DESCRIPTION
**Pull request by albertcastaned**
#### What does this PR do?
- Changes anonymous questions creation to change is_public field to false
- Adds validation/constraint on submit button to enforce an assigned employee when anonymous is toggled on
- Modifies redirect rule to go to question detail instead of home only when its an anonymous question
- Changes globalSuccess toast notification to add the question detail url. This toast does not autoclose.But suggestions to make this feedbnack more visible and better is still appreciated
- Refactors listEmployees controller to fix a bug where some empty user entries would cause an error given they do not have the full_name property. Empty/null entries are now skipped.

#### Where should the reviewer start?
Going to the new controller changes mostly

#### How should this be manually tested?
1. Go to question creation and toggle anonymous question on
2. Validate that the submit button is disabled until an assigned employee is set
3. Create the question 
4. Verify that you are redirected to the question detail
5. Verify that a toast message is shown (and does not autclose) with the question detail url
6. Verify that the is_public field for the question is false (you can check this by seeing if the new icon introducced in #161 appears
7. Extra: Verify that the normal flow for non-anonymous question is intact

#### Any background context you want to provide?
None

#### What are the relevant tickets?
Closes #143 
Closes #172 

#### Screenshots

https://user-images.githubusercontent.com/38927620/215862589-4f564658-b780-4907-a0e0-ed0a3a88fdc9.mov

### CODE REVIEW

**Comment by albertcastaned** [comment]
Missed this section of ticket, working on adding to this on PR 

> Add in the “Thins to keep in mind” (You can also use this in the pop-up):
 
>   - For anonymous questions, you have to select a specif person according to the department selected. 
>   - The anonymous questions will not appear in the home page until an admin publish it. 
>   - Although the question has not been published, you will have an answer.r. 
>   - You must save the link of the question.

File: app/components/Notifications/Notifications.jsx [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
 import { ToastContainer, toast } from 'react-toastify';
-import { DEFAULT_TOAST_CONFIG } from 'app/utils/constants';
+import { DEFAULT_TOAST_CONFIG, QUESTION_CREATED_TOAST_CONFIG } from 'app/utils/constants';
 import useGlobalSuccessMessage from 'app/utils/hooks/useGlobalSuccessMessage';
 
 function Notifications() {
   const globalSuccess = useGlobalSuccessMessage();
   const data = useActionData();
 
+  const successAnonMessage = (message, questionUrl) => (
```
Comment (amilcaralex97) [inline app/components/Notifications/Notifications.jsx:11]: Notification keeps poping up after moving to different routes on wizeq is this an expected behavior?

Comment (albertcastaned) [reply app/components/Notifications/Notifications.jsx:11]: It is an odd behaviour of this line, happened before this PR too. I think it is saved for a short period of time so it has this behaviour sometimes for a short period of time? We can investigate this as a new ticket or I can try to look into it in here. What do you think?

Comment (GeorginaFraire) [reply app/components/Notifications/Notifications.jsx:11]: I think we can open a new ticket for this behavior, but it is up to you  😃

**Review by GeorginaFraire — APPROVED** [review]

File: app/components/Notifications/Notifications.jsx (hunk 1 above)
Comment (albertcastaned) [reply app/components/Notifications/Notifications.jsx:11]: I will create this new ticket and we can work on it, if it becomes too frequent/intrusive we can prioritize it 👍

**Pull request merged**